src/runtime/auto-prompt-manager.ts

```diff
@@ -478,7 +479,10 @@ export class AutoPromptManager {
 
     // For subscriptions, skip triggering checks and use the first potential action
     if (this.isSubscription_({autoPromptType})) {
-      if (shouldShowAutoPrompt) {
```

Comment:
Can we combine this with contribution check below? Ideally, I think we can refactor and simplify the overall flow like

1. Custom suppresion logic (filter out survey based on completiion/failures)
2. General suppresion logic (filter out previous dismissed prompts)
3. Pick the first one (return undefined if monetization prompt type so that `promptFn` gets assign with `displayLargePromptFn`

Reply:
I agree this would be more consistent, however I believe we have to wait until all current contribution publishers have a `TYPE_CONTRIBUTION` action.

For contribution publishers without this action, the `shouldShowMonetizationPromptFromFrequencyCap` boolean overrides the potential action: https://github.com/subscriptions-project/swg-js/blob/main/src/runtime/auto-prompt-manager.ts#L504

I think it makes sense to hold off on this refactoring until after all publishers have their monetization action within audience actions, thoughts?

Reply:
Ahh good catch. I think I missed when `contributionIndex = -1`. Holding off on the refactoring sounds good to me. Thanks for explaining!